index.bs

```diff
@@ -249,20 +252,36 @@ and defensive programming which includes:
     enhanced by the possible usage of a sensor, not degraded by its
     absence.
 
-<div class="example">
+<div class="example" id="robust-example">
+    The following snippet illustrates how an Accelerometer sensor can be created
+    in a robust manner.
     <pre highlight="js">
-        try { // No need to feature detect thanks to try..catch block.
-            var sensor = new GeolocationSensor();
-            sensor.start();
-            sensor.onerror = error => gracefullyDegrade(error);
-            sensor.onreading = _ => updatePosition(sensor.latitude, sensor.longitude);
-        } catch(error) {
-            gracefullyDegrade(error);
+        let accelerometer = null;
+        try {
+            accelerometer = new Accelerometer({ frequency: 10 });
+            accelerometer.addEventListener('error', event => handleErrors(event.error));
+            accelerometer.addEventListener('reading', () => reloadOnShake(accelerometer));
+            accelerometer.start();
+        } catch (error) {
+            handleErrors(error);
+        }
+
+        function handleErrors(error) {
+            if (error.name === 'NotAllowedError') {
+                console.log('Permission to access sensor was denied.');
+            } else if (error.name === 'NotReadableError' ) {
+                console.log('Cannot connect to the sensor.');
+            } else if (error.name === 'SecurityError') {
+                console.log('Sensor construction was blocked by the Feature Policy.');
+            } else if (error.name === 'ReferenceError') {
+                console.log('Sensor is not supported by the User Agent.');
+            } else {
+                throw error;
```

Comment:
Why is throwing considered here?

Reply:
If there are other errors that handled in the if / else blocks, I think it is better to re-throw them, rather than supressing or logging.

Reply:
throwing doesn't make it robust though, an error happened and what ever happened you handled it... throwing can end up affecting other systems, like if someone else ends up catching it

Reply:
This is outside of scope of this example. Robust application should not suppress error propagation to avoid unpredictable behavior. I can fix throwing from error handler.

Reply:
ok